csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.Nuget.cs

```diff
@@ -479,27 +579,63 @@ private IEnumerable<string> GetFeeds(string nugetConfig)
                     continue;
                 }
 
-                yield return url;
+                if (!string.IsNullOrWhiteSpace(url))
+                {
+                    yield return url;
+                }
             }
         }
 
-        private HashSet<string> GetAllFeeds(List<FileInfo> allFiles)
+        private (HashSet<string>, HashSet<string>) GetAllFeeds(List<FileInfo> allFiles)
         {
+            IList<string> GetNugetFeeds(string nugetConfig)
+            {
+                logger.LogInfo($"Getting Nuget feeds from '{nugetConfig}'...");
+                return dotnet.GetNugetFeeds(nugetConfig);
+            }
+
+            IList<string> GetNugetFeedsFromFolder(string folderPath)
+            {
+                logger.LogInfo($"Getting Nuget feeds in folder '{folderPath}'...");
+                return dotnet.GetNugetFeedsFromFolder(folderPath);
+            }
+
             var nugetConfigs = GetAllNugetConfigs(allFiles);
-            var feeds = nugetConfigs
-                .SelectMany(GetFeeds)
-                .Where(str => !string.IsNullOrWhiteSpace(str))
+            var explicitFeeds = nugetConfigs
+                .SelectMany(config => GetFeeds(() => GetNugetFeeds(config)))
                 .ToHashSet();
 
-            if (feeds.Count > 0)
+            if (explicitFeeds.Count > 0)
             {
-                logger.LogInfo($"Found {feeds.Count} Nuget feeds in nuget.config files: {string.Join(", ", feeds.OrderBy(f => f))}");
+                logger.LogInfo($"Found {explicitFeeds.Count} Nuget feeds in nuget.config files: {string.Join(", ", explicitFeeds.OrderBy(f => f))}");
             }
             else
             {
                 logger.LogDebug("No Nuget feeds found in nuget.config files.");
             }
-            return feeds;
+
+            // todo: this could be improved.
+            // We don't have to get the feeds from each of the folders from below, it would be enought to check the folders that recursively contain the others.
```

Comment:
So you are thinking of removing all folder paths that are prefixes of other path before calling the CLI?

Reply:
Yes, that would be an option. (But I won't do it for the time being, as it would add extra complexity)